app/controllers/spree/orders_controller.rb

```diff
@@ -27,13 +27,7 @@ class OrdersController < ::BaseController
     def show
       @order = Spree::Order.find_by!(number: params[:id])
 
-      if params.key?("payment_intent")
-        result = ProcessPaymentIntent.new(params["payment_intent"], @order).call!
-        unless result.ok?
-          flash[:error] = "#{I18n.t("payment_could_not_process")}. #{result.error}"
-        end
-        @order.reload
-      end
+      handle_stripe_response
```

Comment:
We could even make this its own route, and a separate controller action, but this is 90% of the way there as far as clarity.